packages/login/src/components/LoginForm/LoginForm.js

```diff
@@ -31,44 +35,51 @@ export class LoginForm extends Component {
     const usernameFocus = !passwordFocus
 
     return (
-      <StyledLoginForm className="login-form">
+      <StyledLoginFormWrapper>
         {this.props.showTitle
           && <React.Fragment>
             <Typography.H5><FormattedMessage id="client.login.form.title"/></Typography.H5>
             <Typography.P><FormattedMessage id="client.login.form.introduction"/></Typography.P>
           </React.Fragment>}
         <form onSubmit={this.handleSubmit.bind(this)}>
-          <div className="form-group dense">
-            <div className="input-group">
-              <span className="input-group-addon"><Icon icon="user"/></span>
-              <input
-                type="text"
-                className="form-control"
+
+          <StatedValue
+            hasValue={!!this.props.username && this.props.username.length > 0}
```

Comment:
same question here as in other PR (why length check)

Reply:
Thanks for question it. Since "double bang operator" is used the length check is obsolete.

Reply:
see 3a5f4e4a59fc0c8a82c4475e57874c0b1bfc85c9